panic: runtine error:

Getting this error when I try to load a legends/plus export.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13d2d80]

goroutine 69 [running]:
github.com/robertjanetzko/LegendsBrowser2/backend/model.(*HistoricalEvent).setId(0x15375e0?, 0xc00018e030?)
	/home/runner/work/LegendsBrowser2/LegendsBrowser2/backend/model/model.go:2775
github.com/robertjanetzko/LegendsBrowser2/backend/model.parseMapPlus[...](0xc0018b6800, 0xc0002cc518?, 0x180db08)
	/home/runner/work/LegendsBrowser2/LegendsBrowser2/backend/model/parse.go:223 +0xdb
github.com/robertjanetzko/LegendsBrowser2/backend/model.parseDfWorldPlus(0xc0018b29f0?, 0x15d6492?)
	/home/runner/work/LegendsBrowser2/LegendsBrowser2/backend/model/model.go:23928 +0x865
github.com/robertjanetzko/LegendsBrowser2/backend/model.Parse({0xc0001720b0, 0xab}, 0xc000010270)
	/home/runner/work/LegendsBrowser2/LegendsBrowser2/backend/model/parse.go:130 +0x52e
github.com/robertjanetzko/LegendsBrowser2/backend/server.loadWorld(0xc00006e680, {0xc0001720b0, 0xab})
	/home/runner/work/LegendsBrowser2/LegendsBrowser2/backend/server/loading.go:128 +0x45
created by github.com/robertjanetzko/LegendsBrowser2/backend/server.loadHandler.ServeHTTP in goroutine 35
	/home/runner/work/LegendsBrowser2/LegendsBrowser2/backend/server/loading.go:103 +0x636

```
This occurs after the legends_plus.xml has loaded if that makes a difference. It was working on a previous world but all new ones have this problem (i.e. since yesterday).

Let me know if you need more info.

--- a/backend/model/parse.go
+++ b/backend/model/parse.go
@@ -220,10 +220,10 @@ func parseMapPlus[T Identifiable](p *util.XMLParser, dest *map[int]T, creator fu
 				log.Fatal(err)
 			}
 			x, err := creator(p, (*dest)[id])
-			x.setId(id)
 			if err != nil {
 				return
 			}
+			x.setId(id)
 			(*dest)[id] = x
 
 		case util.EndElement:

